### PR TITLE
[WIP] Iceqube integration

### PR DESCRIPTION
**Pull request by aronasorman**
First pass integration of iceqube.

TODOS:
- [x] ~~purge all references to django_q. Kolibri should now work on Python 2 and Python 3 using a single pex file.~~
- [x] Pass in settings to save Job storage SQLite path.
- [x] Port over progress updating to iceqube system.
- [ ] Implement cooperative cancelling

Note: I'm mainly opening this so we can have pex files built on buildkite now.

### CODE REVIEW

File: kolibri/utils/cli.py [hunk 1]
```diff
@@ -14,9 +14,9 @@
 
 import kolibri  # noqa
 from kolibri import dist as kolibri_dist  # noqa
-sys.path = [
+sys.path = sys.path + [
     os.path.realpath(os.path.dirname(kolibri_dist.__file__))
-] + sys.path
+]
```
Comment (aronasorman) [inline kolibri/utils/cli.py:19]: This is a major change with unintended repercussions. it's usually better to have standard library be the first priority, but other installed libs are now getting imported first vs. what's in kolibri.dist.

Note that this change won't affect pex files (our main installation method), but will affect whl files.

**Comment by aronasorman** [comment]
Hmm somehow buildkite isn't detecting my new build. Use this pex file for testing: https://buildkite.com/organizations/learningequality/pipelines/kolibri-build-experimentation/builds/33/jobs/6dde6368-228d-4e1f-94bf-5ddd40f9a877/artifacts/aa7a97ee-3e1f-4fc0-b26b-343f2c51f20c

**Comment by aronasorman** [comment]
I think this is ready to merge now.

**Comment by aronasorman** [comment]
The only remaining task now is cancelling jobs, but that's always been finicky anyway, so I don't think there's too much of a regression left.

**Comment by aronasorman** [comment]
fixed the tests

File: kolibri/utils/cli.py (hunk 1 above)
Comment (benjaoming) [reply kolibri/utils/cli.py:19]: Yup, I originally used `sys.append` but didn't think further about it when it was changed.

What's your take on it? I don't like messing with the sys.path at all.. at least only in an non-intrusive way where we append our path.

File: kolibri/content/api.py [hunk 2]
```diff
@@ -7,7 +7,7 @@
 from django.core.cache import cache
 from django.db.models import Q, Sum
 from django.db.models.aggregates import Count
-from future.moves.urllib.parse import parse_qs, urlparse
+from six.moves.urllib.parse import parse_qs, urlparse
```
Comment (benjaoming) [inline kolibri/content/api.py:10]: Oh cool, didn't know six had urllib stuff, awesome!

File: kolibri/content/utils/transfer.py [hunk 3]
```diff
@@ -45,8 +45,11 @@ def __init__(self, source, dest, block_size=2097152, remove_existing_temp_file=T
         self.finalized = False
         self.closed = False
 
-        signal.signal(signal.SIGINT, self._kill_gracefully)
-        signal.signal(signal.SIGTERM, self._kill_gracefully)
+        # TODO (aron): Instead of using signals, have bbq/iceqube add
+        # hooks that the app calls every so often to determine whether it
+        # should shut down or not.
```
Comment (benjaoming) [inline kolibri/content/utils/transfer.py:49]: Isn't that less efficient and error prone? "Every so often" :)

File: kolibri/core/discovery/utils/filesystem/posix.py [hunk 4]
```diff
@@ -30,20 +35,25 @@ def get_drive_list():
...
         MOUNT_PARSER = OSX_MOUNT_PARSER
     else:
         MOUNT_PARSER = LINUX_MOUNT_PARSER
 
+    try:
+        drivelist = subprocess.Popen('mount', shell=True, stdout=subprocess.PIPE)
+        drivelisto, err = drivelist.communicate()
+    except OSError:  # couldn't run `mount`, let's try reading the /etc/mounts listing directly
+        with open("/proc/mounts") as f:
+            drivelisto = f.read()
```
Comment (benjaoming) [inline kolibri/core/discovery/utils/filesystem/posix.py:48]: It doesn't exist on Windows, and I think you mentioned that this part was scoped out for now? Anyways, it's nice to have these things somehow contained.

A patter could for instance be:

```python
    if sys.platform == "darwin":
        parser = darwin_parser
    elif sys.platform == "posix":
        parser = linux_parser
    else:
        parser = windows_parser

# ...

def windows_parser():
     raise NotImplemented("Windows isn't supported")
```

File: kolibri/tasks/api.py [hunk 5]
```diff
@@ -1,66 +1,52 @@
...
 
 class TasksViewSet(viewsets.ViewSet):
-
     permission_classes = (IsDeviceOwnerOnly, )
 
     def list(self, request):
-        tasks_response = [_task_to_response(t) for t in Task.objects.all()]
-        return Response(tasks_response)
+        # tasks_response = [_task_to_response(t) for t in client.all_jobs()]
+        # return Response(tasks_response)
```
Comment (benjaoming) [inline kolibri/tasks/api.py]: Commented out code doesn't explain what it is?

File: kolibri/tasks/api.py [hunk 6]
```diff
@@ -160,55 +143,56 @@ def localdrive(self, request):
...
-    """
 
-    if not task_instance:
+def _job_to_response(job):
+    if not job:
         return {
-            "type": task_type,
-            "status": "PENDING",
+            "type": "remoteimport",
+            "status": "SCHEDULED",
```
Comment (benjaoming) [inline kolibri/tasks/api.py]: You could add these as pseudo-constants in some module?

File: kolibri/utils/server.py [hunk 7]
```diff
@@ -131,12 +103,10 @@ def run_server(port):
...
-    serve_static_dir(
-        settings.CONTENT_DATABASE_DIR, paths.get_content_database_url("/")
-    )
-    serve_static_dir(
-        settings.CONTENT_STORAGE_DIR, paths.get_content_storage_url("/")
-    )
+    serve_static_dir(settings.CONTENT_DATABASE_DIR,
+                     paths.get_content_database_url("/"))
+    serve_static_dir(settings.CONTENT_STORAGE_DIR,
+                     paths.get_content_storage_url("/"))
```
Comment (benjaoming) [inline kolibri/utils/server.py:109]: Is this a style that you prefer? No change here, right?

File: kolibri/content/utils/transfer.py (hunk 3 above)
Comment (aronasorman) [reply kolibri/content/utils/transfer.py:49]: Yes, but there's no other reliable way to kill threads. Shutting down threads has to be cooperative in some way.

Comment (benjaoming) [reply kolibri/content/utils/transfer.py:49]: Ah okay, I see, for the graceful part

An interesting stale bug: http://bugs.python.org/issue14073

**Comment by benjaoming** [comment]
@aronasorman 
Except for the linting, the tests are passing. Did you get the pex builds working? You crossed out the TODO about it?

**Comment by benjaoming** [comment]
@aronasorman

I've tested the .pex on a clean Ubuntu 17.04 (Python 3.5) and I get this error:

```
test@test-VirtualBox:~/Downloads$ python3 kolibri_buildkite_build-1610_1389_kolibri-v0.4.0-beta5-246-g45243af.pex
Traceback (most recent call last):
  File "/home/test/Downloads/kolibri_buildkite_build-1610_1389_kolibri-v0.4.0-beta5-246-g45243af.pex/.bootstrap/_pex/pex.py", line 310, in execute
  File "/usr/lib/python3.5/contextlib.py", line 59, in __enter__
    return next(self.gen)
  File "/home/test/Downloads/kolibri_buildkite_build-1610_1389_kolibri-v0.4.0-beta5-246-g45243af.pex/.bootstrap/_pex/pex.py", line 243, in patch_sys
  File "/home/test/Downloads/kolibri_buildkite_build-1610_1389_kolibri-v0.4.0-beta5-246-g45243af.pex/.bootstrap/_pex/pex.py", line 202, in minimum_sys
  File "/home/test/Downloads/kolibri_buildkite_build-1610_1389_kolibri-v0.4.0-beta5-246-g45243af.pex/.bootstrap/_pex/pex.py", line 139, in minimum_sys_modules
TypeError: '_NamespacePath' object does not support indexing
```

The issue seems to be described here: `https://github.com/pantsbuild/pex/issues/337`

I'm pretty sure it's not related to any of your changes, although I don't get that it wouldn't have been caught on a previous occasion? Could you just confirm/disconfirm that, and I'll open up a separate issue about it?

**Comment by aronasorman** [comment]
Oh sorry @benjaoming, now they're finally fixed. Last commit makes it so that a  single pex file will work for both python 2 and python 3.

File: kolibri/tasks/api.py (hunk 6 above)
Comment (aronasorman) [reply kolibri/tasks/api.py]: Will fix this.

**Comment by aronasorman** [comment]
@benjaoming I've been using python 3.6.1 and it's working on my side. Maybe it's from that specific version of pex that you have pulled in?

**Comment by rtibbles** [comment]
One more linting issue, @aronasorman!

**Comment by benjaoming** [comment]
Tried the latest .pex again on a clean Ubuntu 17.04 install, same error. Trying other builds.

**Comment by aronasorman** [comment]
Hmm strange @benjaoming -- I have a clean Ubuntu 17.04 install as well, and it's working. Although this is the Vagrantfile file I have set up for Kolibri, which installs some extra things like `build-essential` and `bcc-tools`.

**Comment by aronasorman** [comment]
Are you installing anything extra on top of that installation?

**Comment by benjaoming** [comment]
Opened up #1710, it's not related to this, I can reproduce in other builds, too.

I've installed nothing extra but the Virtualbox guest tools.

**Pull request merged**